src/medida/timer.cc

```diff
@@ -251,14 +257,13 @@ void Timer::Impl::Update(std::chrono::nanoseconds duration) {
 }
 
 
-stats::Snapshot Timer::Impl::GetSnapshot() const {
-  auto values = histogram_.GetSnapshot().getValues();
-  std::vector<double> converted;
-  converted.reserve(values.size());
-  for (auto& v : values) {
-    converted.push_back(v / (double)duration_unit_nanos_);
-  }
-  return {converted};
+stats::Snapshot Timer::Impl::GetSnapshot(uint64_t divisor) const {
```

Comment:
Oh no, this is a mess! I do not like this at all. Sorry, I didn't realize `GetSnapshot` came from `SamplingInterface` as well. Here is my recommended approach:

  - Leave `SamplingInterface::GetSnapshot()` as-is: do not add a `divisor` argument to it.
  - Leave the signature of `Timer::GetSnapshot` and `Timer::Impl::GetSnapshot` alone too: no new `divisor` argument.
  - Have _two_ methods on `Histogram`: both `Histogram::GetSnapshot() const override` which is Histogram's implementation of `SamplingInterface::GetSnapshot() const` as well as an _additional_ method `Histogram::GetSnapshot(divisor=1) const` which is a method _specific_ to `Histogram`. Same name but otherwise no relationship to the method on `SamplingInterface`.
  - Have `Timer::Impl::GetSnapshot()` call `histogram_.GetSnapshot(duration_unit_nanos_.count())` here like you're doing. Timer knows it has a `Histogram`, not just any old `SamplingInterface`, so it can call the `Histogram`-specific method that takes a `divisor` on its `histogram_` member.

Reply:
Aha! This made the code much less messier :) Thanks!